volatility3/framework/plugins/regexscan.py

```diff
@@ -46,38 +46,50 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
-    def _generator(self, regex_pattern):
-        regex_pattern = bytes(regex_pattern, "UTF-8")
-        vollog.debug(f"RegEx Pattern: {regex_pattern}")
+    def _generator(self, layer, pattern, maxsize):
+        vollog.debug(f"RegEx Pattern: {pattern}")
+
+        # Convert string pattern to bytes for RegExScanner
+        pattern_bytes = pattern.encode("utf-8")
+
+        # Compile the pattern here to ensure consistency
+        try:
+            compiled_pattern = re.compile(pattern_bytes)
+        except re.error as e:
+            vollog.error(f"Invalid regex pattern: {e}")
+            raise ValueError(f"Invalid regex pattern: {e}")
 
-        layer = self.context.layers[self.config["primary"]]
         for offset in layer.scan(
-            context=self.context, scanner=scanners.RegExScanner(regex_pattern)
+            context=self.context, scanner=scanners.RegExScanner(pattern_bytes)
```

Comment:
```suggestion
            context=context, scanner=scanners.RegExScanner(pattern_bytes)
```
This should be context